query/src/main/java/org/infinispan/query/impl/LifecycleManager.java

```diff
@@ -121,11 +117,12 @@ public void cacheStarting(ComponentRegistry cr, Configuration cfg, String cacheN
          boolean isIndexed = cfg.indexing().index().isEnabled();
          AdvancedCache<?, ?> cache = cr.getComponent(Cache.class).getAdvancedCache();
 
+         ClassLoader aggregatedClassLoader = makeAggregatedClassLoader(cr.getGlobalComponentRegistry().getGlobalConfiguration().classLoader());
```

Comment:
I've been looking at the `makeAggregatedClassLoader()` implementation, and I really think we should stop adding the TCCL (we already use the TCCL as the default configuration classloader) and the system classloader (all the others delegate to it anyway).

Reply:
Does that hurt or it is just something extra that we can remove?